### DISCUSSION TITLE
Compiler language choice

### DISCUSSION THREAD

**Discussion by zwanzika-hatzel** [carp-lang/Carp#1465] [General]
This is just a question I have out of curiosity. The maker of Carp clearly likes lisps. Are there any technical or preference reasons why the compiler is written in Haskell rather than a lisp? I noticed a lot of compilers tend to prefer languages in the ML family such as Haskell and Ocaml. Why is that? Is lisp inherently ill-suited to write compilers? is it because of the dynamic typing? 
Many thanks

**Comment by eriksvedang** [comment]
Good question... I guess at the time I was extra much into Haskell, and it has a lot of niceties when implementing a language (e.g. parser combinator libraries). The influence of Haskell on Carp can be seen all over the place, for sure.
